File: apps/rust/src/domain/types/fizz_buzz_type03.rs
use crate::domain::model::FizzBuzzValue;

use super::FizzBuzzType;

pub struct FizzBuzzType03;

impl FizzBuzzType for FizzBuzzType03 {
    fn generate(&self, number: i32) -> FizzBuzzValue {
        let value = match (number % 3, number % 5) {
            (0, 0) => "FizzBuzz".to_string(),
            (0, _) => "Fizz".to_string(),
            (_, 0) => "Buzz".to_string(),
            _ => String::new(),
        };
        FizzBuzzValue::new(number, value)
    }
}

File: apps/ruby/lib/fizz_buzz/domain/type/fizz_buzz_type_03.rb
# frozen_string_literal: true

class FizzBuzzType03 < FizzBuzzType
  def generate(number)
    return FizzBuzzValue.new('FizzBuzz', number) if fizz?(number) && buzz?(number)

    FizzBuzzValue.new(number.to_s, number)
  end
end
